File: go/remove-duplicates-sorted-list/remove-duplicates-sorted-list.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	curr := head
	for curr != nil && curr.Next != nil{
		if curr.Val == curr.Next.Val{
			curr.Next = curr.Next.Next
		}else{
			curr = curr.Next
		}
	}
	return head
}

func main() {
	res := deleteDuplicates(
		&ListNode{Val: 1, Next: &ListNode{Val: 1, Next: &ListNode{Val: 2, Next: &ListNode{}}}},
		
	)

	fmt.Println("answer to the first example is:", res.Val, res.Next.Val)
}

File: js/remove-duplicates-sorted-list/remove-duplicates-sorted-list.js
function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

function deleteDuplicates(head){
    let res = head

    while (head && head.next) {
        if (head.val === head.next.val) {
            head.next = head.next.next
        } else {
            head = head.next
        }
    }

    return res 
}

let node3Head = new ListNode(2)
let node2Head = new ListNode(1, node3Head)
let node1Head = new ListNode(1, node2Head)


let res = deleteDuplicates(node1Head)

console.log("answer to the first example is:", res)